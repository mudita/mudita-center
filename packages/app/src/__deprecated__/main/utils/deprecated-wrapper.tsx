/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
