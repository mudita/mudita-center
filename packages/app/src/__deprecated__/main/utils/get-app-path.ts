/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import path from "path"
import packageInfo from "../../../../package.json"

export enum AppType {
  Main = "main",
  Renderer = "renderer",
}
export type AppResolver = () => {
  type: AppType
  app: { getPath: (arg0: string) => string }
}

export const resolve: AppResolver = () => {
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const isRenderer = require("is-electron-renderer")
  // FIXME: Using remote module in renderer process isn't a good pattern.
  //  You can read more in https://github.com/electron/remote#migrating-from-remote
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const app = isRenderer
    ? // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("@electron/remote").app
    : // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("electron").app
  const type = isRenderer ? AppType.Renderer : AppType.Main
  return {
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    app,
    type,
  }
}

const getAppPath = (): string => {
  try {
    return path.join(resolve().app.getPath("appData"), packageInfo.name)
  } catch {
    return ""
  }
}

export default getAppPath
