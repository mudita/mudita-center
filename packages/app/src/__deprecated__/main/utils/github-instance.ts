/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import axios from "axios"

// It's required only for development when API rate limits may exceed
// https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting
const githubToken = process.env.OS_UPDATE_SERVER_ACCESS_TOKEN

export const githubInstance = axios.create({
  headers: {
    ...(githubToken ? { Authorization: `token ${githubToken}` } : {}),
  },
})
