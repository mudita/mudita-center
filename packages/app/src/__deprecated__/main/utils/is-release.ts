/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import isVersionMatch from "App/overview/helpers/is-version-match"

export const isRelease = (version: string): boolean => {
  if(isVersionMatch(version)){
    return /(\d+\.){2}\d+$/i.test(version)
  } else {
    return false
  }
}
