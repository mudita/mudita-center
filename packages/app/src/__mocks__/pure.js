/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

jest.mock("pure", () => ({
  ...jest.requireActual("pure"),
  default: {},
}))

const pure = require("pure")

module.exports = pure
