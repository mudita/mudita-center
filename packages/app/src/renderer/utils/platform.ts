/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export class Platform {
  public macOs(): boolean {
    return process.platform === "darwin"
  }

  public linux(): boolean {
    return process.platform === "linux"
  }

  public windows(): boolean {
    return process.platform === "win32"
  }
}

export const platform = new Platform()
