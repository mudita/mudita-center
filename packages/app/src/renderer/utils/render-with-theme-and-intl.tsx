/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
