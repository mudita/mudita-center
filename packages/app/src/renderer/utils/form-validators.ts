/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { RegisterOptions } from "react-hook-form/dist/types"
import { intl } from "Renderer/utils/intl"

export const phoneNumberRegexp = /^(\+?)(\d(\s?\d)+)$/im

export const primaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value === fields.secondaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const secondaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value === fields.primaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const emailValidator: RegisterOptions = {
  pattern: {
    value:
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
    message: intl.formatMessage({ id: "component.formErrorInvalidEmail" }),
  },
  required: intl.formatMessage({ id: "component.formErrorRequiredEmail" }),
}
