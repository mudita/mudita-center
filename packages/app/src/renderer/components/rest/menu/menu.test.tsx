/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import "@testing-library/jest-dom"
import createMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import React from "react"
import { Router } from "react-router"
import { Provider } from "react-redux"
import { DeviceType } from "@mudita/pure"
import Menu from "Renderer/components/rest/menu/menu.component"
import history from "Renderer/routes/history"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

const mockStore = createMockStore([thunk])

describe("Device: Mudita pure", () => {
  test("matches snapshot", () => {
    const { container } = renderWithThemeAndIntl(
      <Provider
        store={mockStore({
          device: {
            deviceType: DeviceType.MuditaPure,
          },
        })}
      >
        <Router history={history}>
          <Menu />
        </Router>
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})

describe("Device: Mudita harmony", () => {
  test("matches snapshot", () => {
    const { container } = renderWithThemeAndIntl(
      <Provider
        store={mockStore({
          device: {
            deviceType: DeviceType.MuditaHarmony,
          },
        })}
      >
        <Router history={history}>
          <Menu />
        </Router>
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
