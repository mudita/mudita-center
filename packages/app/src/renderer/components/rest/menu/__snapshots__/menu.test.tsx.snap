// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device: Mudita harmony matches snapshot 1`] = `
.c8 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
  color: #6a6a6a;
}

.c11 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2rem;
  width: 2rem;
  position: relative;
  font-style: normal;
}

.c2 svg {
  width: 100%;
  height: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3.2rem;
  width: 3.2rem;
  position: relative;
  font-style: normal;
}

.c6 svg {
  width: 100%;
  height: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 4rem;
  padding: 0.8rem;
  border: none;
  border-radius: 0.2rem;
  color: #6a6a6a;
  font-weight: 400;
  width: 100%;
}

.c5.active {
  background-color: #f4f5f6;
}

.c5.active * {
  color: #000000;
}

.c5.active svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c5:hover {
  background-color: #f4f5f6;
}

.c5:hover * {
  color: #000000;
}

.c5:hover svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c5 svg {
  opacity: 0.75;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 4rem;
  padding: 0.8rem;
  border: none;
  border-radius: 0.2rem;
  color: #6a6a6a;
  font-weight: 400;
  width: 100%;
}

.c12.active {
  background-color: #f4f5f6;
}

.c12.active * {
  color: #000000;
}

.c12.active svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c12:hover {
  background-color: #f4f5f6;
}

.c12:hover * {
  color: #000000;
}

.c12:hover svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c12 svg {
  opacity: 0.75;
}

.c7 {
  margin: 0 0.8rem 0 0;
}

.c9 {
  white-space: nowrap;
}

.c13 {
  padding: 0.4rem 0.8rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 2.4rem 0 1.6rem 0;
}

.c4 {
  margin: 0 1rem 0.4rem 1rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 3.2rem;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  background-color: #ffffff;
}

.c3 {
  height: 2rem;
  width: 8.6rem;
  margin: 2rem 0 2.4rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2 c3"
        data-testid="icon-MuditaLogoWithText"
      >
        <test-file-stub
          image="test-file-stub"
        />
      </span>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        displaystyle="8"
        href="#/news"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuNews"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.news
        </p>
      </a>
    </div>
    <div
      class="c10"
      data-testid="component.menuHeaderDesktopApp"
    >
      <p
        class="c11"
      >
        [value] component.menuHeaderDesktopApp
      </p>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        data-testid="tethering-link"
        displaystyle="8"
        href="#/tethering"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuTethering"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.tethering
        </p>
      </a>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        displaystyle="8"
        href="#/settings"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuSettings"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.settings
        </p>
      </a>
    </div>
    <div
      class="c4"
    >
      <button
        class="c12 c13"
        data-testid="help-menu-button"
        size="2"
        type="button"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuHelp"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.help
        </p>
      </button>
    </div>
  </div>
</div>
`;

exports[`Device: Mudita pure matches snapshot 1`] = `
.c8 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
  color: #6a6a6a;
}

.c11 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2rem;
  width: 2rem;
  position: relative;
  font-style: normal;
}

.c2 svg {
  width: 100%;
  height: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3.2rem;
  width: 3.2rem;
  position: relative;
  font-style: normal;
}

.c6 svg {
  width: 100%;
  height: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 4rem;
  padding: 0.8rem;
  border: none;
  border-radius: 0.2rem;
  color: #6a6a6a;
  font-weight: 400;
  width: 100%;
}

.c5.active {
  background-color: #f4f5f6;
}

.c5.active * {
  color: #000000;
}

.c5.active svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c5:hover {
  background-color: #f4f5f6;
}

.c5:hover * {
  color: #000000;
}

.c5:hover svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c5 svg {
  opacity: 0.75;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 4rem;
  padding: 0.8rem;
  border: none;
  border-radius: 0.2rem;
  color: #6a6a6a;
  font-weight: 400;
  width: 100%;
}

.c12.active {
  background-color: #f4f5f6;
}

.c12.active * {
  color: #000000;
}

.c12.active svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c12:hover {
  background-color: #f4f5f6;
}

.c12:hover * {
  color: #000000;
}

.c12:hover svg {
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
}

.c12 svg {
  opacity: 0.75;
}

.c7 {
  margin: 0 0.8rem 0 0;
}

.c9 {
  white-space: nowrap;
}

.c13 {
  padding: 0.4rem 0.8rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 2.4rem 0 1.6rem 0;
}

.c4 {
  margin: 0 1rem 0.4rem 1rem;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 3.2rem;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  background-color: #ffffff;
}

.c3 {
  height: 2rem;
  width: 8.6rem;
  margin: 2rem 0 2.4rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2 c3"
        data-testid="icon-MuditaLogoWithText"
      >
        <test-file-stub
          image="test-file-stub"
        />
      </span>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        displaystyle="8"
        href="#/news"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuNews"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.news
        </p>
      </a>
    </div>
    <div
      class="c10"
      data-testid="component.menuHeaderDesktopApp"
    >
      <p
        class="c11"
      >
        [value] component.menuHeaderDesktopApp
      </p>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        data-testid="tethering-link"
        displaystyle="8"
        href="#/tethering"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuTethering"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.tethering
        </p>
      </a>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        displaystyle="8"
        href="#/settings"
        size="2"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuSettings"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.settings
        </p>
      </a>
    </div>
    <div
      class="c4"
    >
      <button
        class="c12 c13"
        data-testid="help-menu-button"
        size="2"
        type="button"
      >
        <span
          class="c6 c7"
          data-testid="icon-MenuHelp"
          height="3.2"
          width="3.2"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </span>
        <p
          class="c8 c9"
        >
          [value] module.help
        </p>
      </button>
    </div>
  </div>
</div>
`;
